asyncio_run_in_process/run_in_process.py

```diff
@@ -185,10 +184,17 @@ def open_in_process(
     loop: asyncio.AbstractEventLoop = None,
     subprocess_kwargs: 'SubprocessKwargs' = None,
 ) -> AsyncContextManager[ProcessAPI[TReturn]]:
-    return cast(
-        AsyncContextManager[ProcessAPI[TReturn]],
-        _open_in_process(async_fn, *args, loop=loop, subprocess_kwargs=subprocess_kwargs),
-    )
+    return _open_in_process(
+        async_fn, *args, loop=loop, subprocess_kwargs=subprocess_kwargs, use_trio=False)
+
+
+def open_in_process_with_trio(
```

Comment:
I'm :+1: on exposing this type of dedicated shortcut.  I would also be supportive of modifying the main `run_in_process` and `open_in_process` APIs to support something like `run_in_process(..., engine='trio')`.  That would also imply exposing a `run_in_process_with_asyncio` top level export as well for API symmetry.

Reply:
I didn't want to expose APIs that require explicitly specifying the engine because that'd cause us to have mutually exclusive arguments (the loop must not be passed when using trio), and also because it'd be easier to accidentally misuse the default API to run trio code.